### DISCUSSION TITLE
Annotations (highlights) don't save to the PDF

### DISCUSSION THREAD

**Discussion by hrrrps** [sumatrapdfreader/sumatrapdf#3717] [Q&A] [answered]
Using Sumatra PDF on a Windows 11 machine.

Anyone else have this problem? Solutions?

**Comment by kjk** [comment]
Which version of SumatraPDF?

What exactly happens?

Does it happen with every PDF or just a specific one?

Does it still happen in pre-release version: https://www.sumatrapdfreader.org/prerelease ? If yes, please open an issue and provide the file that shows the problem.

**Reply by hrrrps** [reply]
v3.4.6.
all PDFs

I make an annotation (cursor selection + a), it highlights, I save the file, replacing the existing copy, I return to the document and the annotation has disappeared.

I will try the prerelease.

**Reply by hrrrps** [reply]
this is a sample file which a tried to annotate with the prelease. same problem persists.

[diser-privacy-policy.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/12708998/diser-privacy-policy.pdf)

**Reply by kjk** [reply, marked as answer]
I can't reproduce this. I tried:
* open the file
* create a higlight annotation with `a`
* save annotations to the same file with `Ctrl + Shift + S`
* close file
* re-open the file

The annotation is there.

Please provide full reproduction steps, preferably with a captured video to show the problem.

**Reply by kjk** [reply]
also make sure that the file is not marked as read only (although this should generate an error notification when saving)

**Reply by GitHubRulesOK** [reply]
@kjk there is a potential for user to add annots and instinctively use **main file save as,** I dont think that add annots and perhaps should not but perhaps needs warning?

**Reply by kjk** [reply]
Good point but more likely I'll just remove Save As. There's a lot of problems with it and I would rather not maintain this code in this form.

**Reply by hrrrps** [reply]
The Ctrl + Shift + S command worked. I was using Ctrl +  S then saving via the file explorer. I have a video of what that all looks like - I just need to find a way to blur parts of it before uploading.

Thanks for being so responsive. I like having a FOSS alternative to Adobe. I will be donating to your project.

**Reply by kjk** [reply]
Thanks, no need for the video. The crucial info is that you were using `Ctrl + S` which didn't save the annotations. I just fixed it in https://github.com/sumatrapdfreader/sumatrapdf/commit/95ebef7f936b65cb0f82a6e455bf4043890ca84e so in the next pre-release builds `Ctrl + S` should also save annotations.

**Reply by GitHubRulesOK** [reply]
@kjk I depend heavily on "Save As" for convert CBZ **etc** to PDF  I would be sad to see that go even if it has warts.

**Comment by ggatech** [comment]
The "Save Changes to a new PDF" option in the annotations editor (upper-case A after selecting text) does save the annotations when saving to a new document name.
